marlowe/src/Language/Marlowe/SemanticsTypes.hs

```diff
@@ -184,9 +186,13 @@ data Payee = Account AccountId
     datatype Case is mutually recurvive with @Contract@
 -}
 data Case a = Case Action a
+            | MerkleizedCase Action BuiltinByteString
```

Comment:
Sometimes the practice in other Cardano packages is to create a `newtype` and a `Hash` type-family member, but I don't know that Marlowe should conform to that.
```haskell
newtype SerialisedContinuation = SerialisedContinuation {unSerialisedContinuation :: BuiltinByteString}
data Case a = Case Action a | MerkleizedCase Action (Hash SerialisedContinuation)
data Input = NormalInput InputContent | MerkleizedInput InputContent SerialisedContinuation
```
In any case, it seems safer to use types to distinguish the continuation from its hash.

Reply:
This makes a lot of sense, but I am going to leave it out for now since we no longer have to deal with both serialised continuations and hashes